pkg/sdk/go/things_test.go

```diff
@@ -1404,20 +1404,72 @@ func TestShareThing(t *testing.T) {
 		},
 	}
 
+	for _, tc := range cases {
+		repoCall := pRepo.On("EvaluateGroupAccess", mock.Anything, mock.Anything).Return(policies.Policy{}, nil)
+		repoCall1 := pRepo.On("Retrieve", mock.Anything, mock.Anything).Return(policies.PolicyPage{}, nil)
+		repoCall2 := pRepo.On("Save", mock.Anything, mock.Anything).Return(policies.Policy{}, nil)
+		err := mfsdk.ShareThing(tc.groupID, tc.userID, []string{"c_list", "c_delete"}, tc.token)
+		assert.Equal(t, tc.err, err, fmt.Sprintf("%s: expected error %s, got %s", tc.desc, tc.err, err))
+		if tc.err == nil {
+			ok := repoCall2.Parent.AssertCalled(t, "Save", mock.Anything, mock.Anything)
+			assert.True(t, ok, fmt.Sprintf("Save was not called on %s", tc.desc))
+		}
+		repoCall.Unset()
+		repoCall1.Unset()
+		repoCall2.Unset()
+	}
+
+	aPolicy = cmocks.SubjectSet{Subject: "things", Relation: []string{}}
+	uauth = cmocks.NewAuthService(users, map[string][]cmocks.SubjectSet{adminID: {aPolicy}})
+
+	psvc = policies.NewService(uauth, pRepo, policiesCache, idProvider)
+
+	svc = clients.NewService(uauth, psvc, cRepo, gRepo, thingCache, idProvider)
+	ts = newThingsServer(svc, psvc)
+	defer ts.Close()
+
+	conf = sdk.Config{
+		ThingsURL: ts.URL,
+	}
+	mfsdk = sdk.NewSDK(conf)
+
+	cases = []struct {
```

Comment:
No need for separate cases here, you can add these to the `cases` slice above.

Reply:
Something like [this](https://github.com/dborovcanin/mainflux/commit/9625266d617398c9d603a5832bfe8594ba32ebb1).